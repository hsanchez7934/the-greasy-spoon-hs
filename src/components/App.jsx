import React, { Component } from 'react';
import './App.css';
import apiKey from '../apikey.js';

class App extends Component {

  componentDidMount() {
    fetch(`https://check-api.herokuapp.com/tables`, {
      headers: {
        Authorization: apiKey
      }
    })
    .then(response => response.json())
    .then(response => console.log(response))
  }

  render() {
    return (
      <div className="App">
        hello world
      </div>
    );
  }
}

export default App;
