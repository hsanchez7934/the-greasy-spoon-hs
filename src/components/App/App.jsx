import React, { Component } from 'react';
import './App.css';
import apiKey from '../../apikey.js';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
import HomeScreen from '../HomeScreen/HomeScreen.jsx';

class App extends Component {

  componentDidMount() {
    // this.getTables();
      // this.getChecks();
  }


  getTables = () => {
    fetch(`https://check-api.herokuapp.com/checks`, {
      method: 'DELETE',
      headers: {
        Authorization: apiKey
      }
    })
      .then(response => response.json())
      .then(response => console.log(response))
      .catch(error => console.log(error));
  };

  getChecks = () => {
    fetch(`https://check-api.herokuapp.com/checks/1e6df0a0-50b7-46a0-968a-961bab96a0ae`, {
      headers: {
        Authorization: apiKey
      }
    })
      .then(response => response.json())
      .then(response => console.log(response))
      .catch(error => console.log(error));
  };

  render() {
    return (
      <div className="App">
        <HomeScreen />
      </div>
    );
  }
}

export default App;
