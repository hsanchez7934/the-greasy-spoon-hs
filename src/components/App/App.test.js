import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { mount } from 'enzyme';
import config from '../../tests/testSetup.js';
import toJson from 'enzyme-to-json';

describe(`App component unit test`, () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<App />);
  });

  test(`should render an instance of App`, () => {
    expect(wrapper.exists()).toEqual(true);
  });

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test(`should render correctly`, () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test(`should have parent div node with class name App`, () => {
    const parent = wrapper.find('div').at(0);

    expect(parent.exists()).toEqual(true);
    expect(parent.hasClass('App')).toEqual(true);
  });

  test(`should render an instance of HomeScreen component `, () => {
    const homescreen = wrapper.find('HomeScreen');
    expect(homescreen.exists()).toEqual(true);
  });
  
});
