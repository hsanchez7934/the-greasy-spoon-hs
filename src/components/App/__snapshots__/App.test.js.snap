// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component unit test should render correctly 1`] = `
<App>
  <div
    className="App"
  >
    <HomeScreen>
      <BrowserRouter>
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <section
            id="homescreen-container"
          >
            <nav>
              <section
                id="nav-left"
              >
                <img
                  alt="Mobile menu icon."
                  className="mobile-menu-icon"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  src="menu.svg"
                />
              </section>
              <section
                id="nav-right"
              >
                <h6
                  id="mobile-title"
                >
                  The Greasy Spoon
                </h6>
              </section>
            </nav>
            <div
              className="drop-down-menu"
            >
              <div
                id="cancel-icon-container"
              >
                <img
                  alt="Mobile menu icon."
                  className="mobile-menu-icon"
                  id="cancel-icon"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  src="cancel.svg"
                />
              </div>
              <ul
                id="nav-links-ul"
              >
                <li
                  className="nav-links-li"
                >
                  <Link
                    className="nav-links-link-tag"
                    replace={false}
                    to="/"
                  >
                    <a
                      className="nav-links-link-tag"
                      href="/"
                      onClick={[Function]}
                    >
                      Home
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-links-li"
                >
                  <Link
                    className="nav-links-link-tag"
                    replace={false}
                    to="/tables"
                  >
                    <a
                      className="nav-links-link-tag"
                      href="/tables"
                      onClick={[Function]}
                    >
                      Tables
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-links-li"
                >
                  <Link
                    className="nav-links-link-tag"
                    replace={false}
                    to="/openchecks"
                  >
                    <a
                      className="nav-links-link-tag"
                      href="/openchecks"
                      onClick={[Function]}
                    >
                      Open Checks
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-links-li"
                >
                  <Link
                    className="nav-links-link-tag"
                    replace={false}
                    to="/closedchecks"
                  >
                    <a
                      className="nav-links-link-tag"
                      href="/closedchecks"
                      onClick={[Function]}
                    >
                      Closed Checks
                    </a>
                  </Link>
                </li>
              </ul>
            </div>
            <aside
              id="sidebar-container"
            >
              <section
                id="restaurant-title"
              >
                <h1
                  id="greasy-spoon-text"
                >
                  The Greasy Spoon
                </h1>
              </section>
              <section
                id="links-container"
              >
                <ul
                  id="nav-links-ul"
                >
                  <li
                    className="nav-links-li"
                  >
                    <Link
                      className="nav-links-link-tag"
                      replace={false}
                      to="/"
                    >
                      <a
                        className="nav-links-link-tag"
                        href="/"
                        onClick={[Function]}
                      >
                        Home
                      </a>
                    </Link>
                  </li>
                  <li
                    className="nav-links-li"
                  >
                    <Link
                      className="nav-links-link-tag"
                      replace={false}
                      to="/tables"
                    >
                      <a
                        className="nav-links-link-tag"
                        href="/tables"
                        onClick={[Function]}
                      >
                        Tables
                      </a>
                    </Link>
                  </li>
                  <li
                    className="nav-links-li"
                  >
                    <Link
                      className="nav-links-link-tag"
                      replace={false}
                      to="/openchecks"
                    >
                      <a
                        className="nav-links-link-tag"
                        href="/openchecks"
                        onClick={[Function]}
                      >
                        Open Checks
                      </a>
                    </Link>
                  </li>
                  <li
                    className="nav-links-li"
                  >
                    <Link
                      className="nav-links-link-tag"
                      replace={false}
                      to="/closedchecks"
                    >
                      <a
                        className="nav-links-link-tag"
                        href="/closedchecks"
                        onClick={[Function]}
                      >
                        Closed Checks
                      </a>
                    </Link>
                  </li>
                </ul>
              </section>
            </aside>
            <div
              id="container2"
            >
              <Route
                component={[Function]}
                exact={true}
                key="0"
                path="/"
              >
                <main
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                >
                  <HomePage>
                    <section
                      id="homepage-container"
                    >
                      <section
                        id="homepage-information-container"
                      >
                        <h1
                          id="homepage-title"
                        >
                          The Greasy Spoon
                        </h1>
                        <h2
                          id="homepage-welcome"
                        >
                          Welcome
                        </h2>
                        <p
                          id="homepage-text"
                        >
                          We will be placing orders for our valued customers using this application. The sidebar located on the left will help you to easily navigate the app.  The Tables link will show you which tables are are available to open a check against. Click on the Open Checks and Closed Checks links to view open and closed checks. Happy serving!
                        </p>
                        <div
                          id="get-started-button-container"
                        >
                          <Link
                            id="get-started-button-link-tag"
                            replace={false}
                            to="/tables"
                          >
                            <a
                              href="/tables"
                              id="get-started-button-link-tag"
                              onClick={[Function]}
                            >
                              <button
                                id="get-started-button"
                              >
                                GET STARTED
                              </button>
                            </a>
                          </Link>
                        </div>
                      </section>
                    </section>
                  </HomePage>
                </main>
              </Route>
              <Route
                component={[Function]}
                key="1"
                path="/tables"
              />
              <Route
                component={[Function]}
                key="2"
                path="/currentcheck"
              />
              <Route
                component={[Function]}
                key="3"
                path="/openchecks"
              />
              <Route
                component={[Function]}
                key="4"
                path="/closedchecks"
              />
            </div>
          </section>
        </Router>
      </BrowserRouter>
    </HomeScreen>
  </div>
</App>
`;
